customerio/customerio-reactnative
# feat: fetch currently stored device token using JS method

Closes: https://github.com/customerio/issues/issues/11181

Provides a feature to fetch currently stored device token and returns as a string in a promise. This feature helps getting device tokens for FCM & APN for iOS and FCM token for Android. The following is the use case of the feature:

```
   CustomerIO.pushMessaging()
      .getRegisteredDeviceToken()
      .then((token) => {
        setDeviceToken(token);
      })
      .catch((error) => {
        console.log(error);
   });
```

### How to test this feature?
- Open the **Sample app**.
- Go to the **Settings** screen.
- Look at the very first field, which is labeled **Device token**.
- If the app has generated a device token, you will see it in this field. Otherwise, the field will be blank.
